util/src/main/java/tech/pegasys/artemis/util/bls/BLSKeyPair.java

```diff
@@ -28,20 +27,15 @@ public static BLSKeyPair random(int entropy) {
     return new BLSKeyPair(KeyPair.random(entropy));
   }
 
-  public BLSKeyPair(BLSPublicKey publicKey, BLSSecretKey secretKey) {
-    this.publicKey = publicKey;
-    this.secretKey = secretKey;
-  }
-
   public BLSKeyPair(KeyPair keyPair) {
-    this(new BLSPublicKey(keyPair.publicKey()), new BLSSecretKey(keyPair.secretKey()));
+    this.keyPair = keyPair;
   }
 
   public BLSPublicKey getPublicKey() {
-    return publicKey;
+    return new BLSPublicKey(keyPair.publicKey());
   }
 
   public BLSSecretKey getSecretKey() {
-    return secretKey;
+    return new BLSSecretKey(keyPair.secretKey());
   }
```

Comment:
It's generally better to avoid getters having to create the object they're returning.  Was there a reason we're now deferring the creation of these `BLSPublicKey` and `BLSSecretKey` objects?

Admittedly I normally don't like constructors creating new objects but given these are just wrappers the benefit of caching them seems to outweigh the slight unpleasantness of constructors doing "work".

Reply:
Yeah, I've reverted and improved this now. Not sure what I was thinking.